src_c/camera_v4l2.c

```diff
@@ -29,42 +29,64 @@ char **
 v4l2_list_cameras(int *num_devices)
 {
     char **devices;
-    char *device;
+    char *device = NULL;
     int num, i, fd;
 
-    num = *num_devices;
+    num = *num_devices = 0;
 
     devices = (char **)malloc(sizeof(char *) * 65);
-
-    device = (char *)malloc(sizeof(char) * 13);
-    strcpy(device, "/dev/video");
-    fd = open(device, O_RDONLY);
-    if (fd != -1) {
-        devices[num] = device;
-        num++;
-        device = (char *)malloc(sizeof(char) * 13);
+    if (!devices) {
+        return NULL;
     }
-    close(fd);
+
     /* v4l2 cameras can be /dev/video and /dev/video0 to /dev/video63 */
-    for (i = 0; i < 64; i++) {
-        sprintf(device, "/dev/video%d", i);
+    for (i = -1; i < 64; i++) {
+        device = (char *)malloc(sizeof(char) * 13);
+        if (!device) {
+            goto error;
+        }
+
+        if (i == -1) {
+            strcpy(device, "/dev/video");
+        }
```

Comment:
Why do you do this? "This" being the i == -1 iteration.

Reply:
This is a special case. `/dev/video` is a thing, and `/dev/video0` to `/dev/video63`. So in total we need 65 iterations to handle all cases.

The previous code handled `/dev/video` outside the loop, but with this, it is handled in the loop for better consistency and easier error handling

Reply:
Oh, I missed that. Thanks.